apps/remote_control/lib/lexical/remote_control/code_mod/ast.ex

```diff
@@ -36,7 +82,11 @@ defmodule Lexical.RemoteControl.CodeMod.Ast do
           within_range?(node, position)
         end)
 
-      {:ok, zipper}
+      if zipper do
+        {:ok, zipper}
+      else
+        {:error, :not_found}
+      end
```

Comment:
it feels like this should go inside the with. Perhaps extract the pipe into a function
```elixir
with {:ok, ast} <- from(document) do 
  zipper_at_position(ast, position)
end

defp zipper_at_position(ast, position) do 
  zipper = 
    ast 
    |> Zipper.zip()
    |> Zipper.find(&within_range?(&1, position)

  if zipper do 
    {:ok, zipper}
  else
    {:error, :not_found}
  end
end
```